sdk/python/feast/feature_store.py

```diff
@@ -745,7 +746,8 @@ def plan(
         # Compute the desired difference between the current infra, as stored in the registry,
         # and the desired infra.
         self._registry.refresh(project=self.project)
-        current_infra_proto = self._registry.proto().infra.__deepcopy__()
+        current_infra_proto = RegistryProto().infra
```

Comment:
You can use Infra proto object directly instead of getting it from RegistryProto

Reply:
Will review this to check. Need to check some behaviors of doing that.

Reply:
so you have to be careful when you call `self._registry.proto()` you're referencing the `FeatureStore` class and you want to make sure that the initialization was done so it could be fetched from the database. 

All of the initialization happens within the `FeatureStore.__init__()` method. 

If this still behaves as expected using that, then I'm fine with this but it doesn't seem like it would but I only briefly looked at this. 

I knew to check this because my team found that this implementation of the registry has some bottlenecks I intend on fixing.

Reply:
CC @tokoko @shuchu

Reply:
Looks to me it should be fine in this scenario. There's even a `self._registry.refresh()` call before. It would have failed then if called on an uninitialized `FeatureStore` object.